cheri_compressed_cap_common.h

```diff
@@ -723,40 +732,27 @@ static inline bool _cc_N(is_representable_new_addr)(bool sealed, _cc_addr_t base
     }
 }
 
-static inline _cc_addr_t _cc_N(cap_bounds_address)(const _cc_cap_t* cap) {
-    // Remove flags bits
-    _cc_addr_t cursor = cap->_cr_cursor & _CC_CURSOR_MASK;
-    // Sign extend
-    if (cursor & ((_CC_CURSOR_MASK >> 1) + 1))
-        cursor |= ~_CC_CURSOR_MASK;
-    return cursor;
-}
-
+#ifdef CC_IS_MORELLO
```

Comment:
I'd much rather see less #ifdef'ing than more; the fact we have CC_IS_MORELLO at all rather than nicer parameterisations makes me a little sad